### PR TITLE
Add operator-observability rules unit test

### PR DESCRIPTION
**Pull request by assafad**
**What this PR does / why we need it**:
Add [operator-observability rules unit test](https://github.com/machadovilaca/operator-observability/tree/main/pkg/testutil), which ensures that alerts and recording rules definitions are following the monitoring best practices, e.g. alerts have all the required annotations and labels with expected values, recording rules have expressions, etc.

Fixes https://issues.redhat.com/browse/CNV-36763
Other PRs references: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2827, https://github.com/kubevirt/kubevirt/pull/11454

/cc @machadovilaca, @oshoval

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/monitoring/rules/rules_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/machadovilaca/operator-observability/pkg/testutil"
+)
+
+func TestRules(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Rules Suite")
+}
```
Comment (machadovilaca) [inline pkg/monitoring/rules/rules_test.go:15]: usually, the convention is to create a `..._suite_test.go` file with this code and a `...test.go` for the actual test code, such as in:

https://github.com/kubevirt/cluster-network-addons-operator/blob/main/pkg/controller/networkaddonsconfig/networkaddonsconfig_controller_suite_test.go

https://github.com/kubevirt/cluster-network-addons-operator/blob/main/pkg/controller/networkaddonsconfig/networkaddonsconfig_controller_test.go

**Review by machadovilaca — CHANGES_REQUESTED** [review]

File: pkg/monitoring/rules/rules_test.go (hunk 1 above)
Comment (assafad) [reply pkg/monitoring/rules/rules_test.go]: Thanks, done.

File: pkg/monitoring/rules/rules_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package rules
```
Comment (oshoval) [inline pkg/monitoring/rules/rules_suite_test.go:1]: please use package rules_test (both files)
all the functions that you are testing are already external so it is easy

**Review by oshoval — COMMENTED** [review]
Thanks

Please consider splitting to commits, one that does the vendoring and one for the code changes, here it is easy to review, but when it isn't it is much better.

Do we need all the PR referenced ? it is nice to have, but maybe lets trim the list please to the most important ones?

File: pkg/monitoring/rules/rules_suite_test.go (hunk 2 above)
Comment (assafad) [reply pkg/monitoring/rules/rules_suite_test.go]: I had to import rules package for using some functions, but done.

**Comment by assafad** [comment]
> Thanks
> 
> Please consider splitting to commits, one that does the vendoring and one for the code changes, here it is easy to review, but when it isn't it is much better.
> 
> Do we need all the PR referenced ? it is nice to have, but maybe lets trim the list please to the most important ones?

@oshoval Done.

**Pull request merged**